Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3901,18 +3895,10 @@ - (void)updateLastVisibleTimestamp
 
 - (void)cleanUpUnreadIndicatorIfNecessary
 {
-    BOOL hasUnreadIndicator = self.dynamicInteractions.unreadIndicatorPosition != nil;
+    BOOL hasUnreadIndicator = self.dynamicInteractions.unreadIndicator != nil;
     if (!hasUnreadIndicator) {
         return;
     }
-    __block BOOL hasUnseenInteractions = NO;
-    [self.uiDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
-        hasUnseenInteractions =
-            [[transaction ext:TSUnreadDatabaseViewExtensionName] numberOfItemsInGroup:self.thread.uniqueId] > 0;
-    }];
-    if (hasUnseenInteractions) {
```

Comment:
I'm not sure of the consequences of this in aggregate - but typically `ensureDynamicInteractions` is quite expensive, and now we've removed a (seemingly quite common) guard that avoids running it. I just want to make sure this isn't going to destroy perf.

Looking a bit closer, it seems like these guards are still in place, but now they happen within the context of a read/write transaction in ensureDynamicInteractions, is that right?

Reply:
I just realized that we can now remove this method `cleanUpUnreadIndicatorIfNecessary` entirely.  Since the unread indicator is no longer an interaction, there's no need to eagerly clean it up.